### PR TITLE
minimal changes to fix issue #248

### PR DESCRIPTION
**Pull request by gidden**
Here are the minimal changes I needed to fix my use case. The enums had to be moved up because of ordering issues (could also add functions to header file). 

Here string checking also checks numpy scalar strings. Then it is also checked when converting to bytes.

As I said, I have not tested this against other issues - perhaps you have CI? I'm happy to respond to any comments you may have.

### CODE REVIEW

File: src/python.cpp [hunk 1]
```diff
@@ -136,7 +136,7 @@ std::string as_std_string(PyObject* str) {
 
   // conver to bytes if its unicode
   PyObjectPtr pStr;
-  if (PyUnicode_Check(str)) {
+  if (PyUnicode_Check(str) | isPyArrayScalar(str)) {
```
Comment (kevinushey) [inline src/python.cpp]: A nitpick, but should we use `||` rather than `|` here?

**Comment by kevinushey** [comment]
This PR looks good to me -- thanks for digging in! I'll let @jjallaire say the final word as he's more familiar with the guts here just in case this change might prompt any other changes we might need around this part of the code.

**Comment by gidden** [comment]
I addressed your comment in the most recent commit @kevinushey. If there are some instructions on how to add tests, I can look into that as well.

File: src/python.cpp [hunk 2]
```diff
@@ -136,7 +136,7 @@ std::string as_std_string(PyObject* str) {
 
   // conver to bytes if its unicode
   PyObjectPtr pStr;
-  if (PyUnicode_Check(str)) {
+  if (PyUnicode_Check(str) || isPyArrayScalar(str)) {
```
Comment (jjallaire) [inline src/python.cpp]: Presumably at this point we don't need an additional check that the PyArrayScalar is in fact a string scaler? (i.e. the callers of as_std_string have already established that the Python object is a string?)

Comment (jjallaire) [reply src/python.cpp]: I reviewed the various calls to `as_std_string()` and it does in fact look like we always know that we have a string.

**Pull request merged**